jupyterlab_git/tests/test_init.py

```diff
@@ -0,0 +1,88 @@
+import os
+from subprocess import CalledProcessError
+from unittest.mock import Mock, call, patch
+
+import pytest
+import tornado
+
+from jupyterlab_git import JupyterLabGit
+from jupyterlab_git.git import Git
+
+from .testutils import FakeContentManager, maybe_future
+
+
+@pytest.mark.asyncio
+async def test_init():
+    with patch("jupyterlab_git.git.execute") as mock_execute:
+        # Given
+        mock_execute.return_value = maybe_future((0, "", ""))
+
+        # When
+        actual_response = await Git(FakeContentManager("/bin")).init("test_curr_path")
+
+        mock_execute.assert_called_once_with(
+            ["git", "init"], cwd=os.path.join("/bin", "test_curr_path")
+        )
+
+        assert {"code": 0, "actions": None} == actual_response
+
+
+@pytest.mark.asyncio
+async def test_init_and_post_init():
+    with patch("jupyterlab_git.git.execute") as mock_execute:
+        # Given
+        mock_execute.side_effect = [
+            maybe_future((0, "", "")),
+            maybe_future((0, "hello", "")),
+        ]
+
+        # When
+        actual_response = await Git(
+            FakeContentManager("/bin"),
+            JupyterLabGit(actions={"post_init": ['echo "hello"']}),
+        ).init("test_curr_path")
+
+        mock_execute.assert_called_with(
+            ["echo", "hello"], cwd=os.path.join("/bin", "test_curr_path")
+        )
+
+        assert {
+            "code": 0,
+            "actions": [
+                {"cmd": 'echo "hello"', "code": 0, "stderr": "", "stdout": "hello"}
+            ],
+        } == actual_response
+
+
+@pytest.mark.asyncio
+async def test_init_and_post_init_fail():
+    with patch("jupyterlab_git.git.execute") as mock_execute:
+        # Given
+        mock_execute.side_effect = [
+            maybe_future((0, "", "")),
+            maybe_future((1, "", "not_there: command not found")),
```

Comment:
You can test the `Exception` emission by duplicating this test and set an Expection object as second element in the list.

Reply:
That tip was a big time saver - tested!